### PR TITLE
Fix date type handling when parsing excel files

### PR DESCRIPTION
**Pull request by vanathi-g**
## Changes:
- Use `DataFormatter` to use set cell format when parsing date values

## Preview of changes:

Sample date formats (excel sheet):

Current behaviour:

Behaviour after fix:

### CODE REVIEW

File: wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java [hunk 1]
```diff
@@ -165,8 +167,8 @@ public List<Row> execute(List<Row> records, final ExecutorContext context)
 
                   case NUMERIC:
                     if (HSSFDateUtil.isCellDateFormatted(cell)) {
-                      newRow.add(name, cell.getDateCellValue());
-                      value = cell.getDateCellValue().toString();
+                      newRow.add(name, formatter.formatCellValue(cell));
```
Comment (iam-divya) [inline wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java]: Is there any test case that need to be updated to capture this scenario?

File: wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java [hunk 2]
```diff
@@ -165,8 +167,8 @@ public List<Row> execute(List<Row> records, final ExecutorContext context)
...
+                      value = formatter.formatCellValue(cell);
```
Comment (iam-divya) [inline wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java:169]: nit: adding this line after value assignment, newRow.add(name, value);

**Review by iam-divya — COMMENTED** [review]
Left comments

File: wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java (hunk 1 above)
Comment (vanathi-g) [reply wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java]: Done, added a test

File: wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java (hunk 2 above)
Comment (vanathi-g) [reply wrangler-core/src/main/java/io/cdap/directives/parser/ParseExcel.java:169]: Cleaned up code

**Review by saimukkamala — APPROVED** [review]

**Pull request merged**